pkg/generator/generator.go

```diff
@@ -53,205 +57,239 @@ type keyGenConfig struct {
 
 // GenKeys load secrets from a secret manager or generate them and save to a secret manager
 // GenKeys generates keys until there's an error or a dependency that can't be set.
-func GenKeys(config *GenConfig, keys []*v1alpha1.KeyConfig) error {
+func (g *GenConfig) GenKeys(ctx context.Context) error {
 	// A queue that will allow re-processing of self referencing keys that exist
-	queue, enqueuer := keyQueue(keys)
-	// generate until queue done
-	for key := range queue {
-		log.V(1).Info("reconciling secret key", "secret_name", secretReq.Name,
-			"data_key", key.Name, "secret_type", string(key.Type))
-
-		keyGenerator, err := generator.NewKeyGenerator(secretReq.Name, instance.Namespace, key, &instance.Spec.AppConfig)
-		if err != nil {
-			log.V(0).Info("error routing secret key type",
-				"secret_name", secretReq.Name,
+	g.Log = g.Log.WithValues("secret_name", g.SecObject.Name, "namespace", g.Namespace)
+	requeue, enquerer := makeKeyQueue(g.KeysToGen)
```

Comment:
maybe just queue instead of enquerer?

Reply:
hmm, no thats not what that variable is.... its a string?

Reply:
`enquerer`, could be `enquererFn` it checks to make sure the self referenced key exits

Reply:
@wstrange I've updated this, take a look.